### ISSUE TITLE
Ideas and thoughts

### ISSUE THREAD

**Issue by mrvollger** [wjwei-handsome/wgatools#1]
Hi @wjwei-handsome,

Thanks for making this tool! This is just a list of thoughts I had looking through the repo, that you might consider. 

* you are much better at rust than I am!
* it looks like you might be reading the whole PAF file into memory, maybe there is a way to iterate record by record instead?
* it would be awesome if the PAF reader could also parse the `cs` tag into a CIGAR string allowing the conversion to other formats
* it would be great to update your io code to allow reading and writing of compressed files. If you want an example you can look here: https://github.com/fiberseq/fibertools-rs/blob/633277099e1fde8684733c05d6b192434ee01c10/bio-io/src/lib.rs#L1-L99
* For visualization might I humbly suggest integration with rustybam and [SafFire](https://mrvollger.github.io/SafFire/#dataset=default&ref=CHM13_v1.1&query=GRCh38)? 

I think that is all my ideas at the moment, but I will add as I think of things!

Thanks again,
Mitchell

**Comment by wjwei-handsome** [comment]
Hi @mrvollger ,

Your encouragement and advice have cheered me up this morning! Thanks!

As for your point:
- point2: I actually implemented a buf-reader and iterator for [paf,maf,chian] records. For PAF example: 
 https://github.com/wjwei-handsome/wgatools/blob/c6342f3de7250e092d50caf8dfda7eb9f3d0e3df/src/parser/paf.rs#L102C1-L112C6
- point3: great idea! I will learn more for `cs` tag
- point4: Of course!  I think it should be simple
- point5: My pleasure! This is gonna be cool

To sum up, this project still needs continuous improvement. At present, it is only a rough framework :)
I will try my best to improve it!

Best wishes,
Wei
